### ISSUE TITLE
rmd_reader and figures not compatible with develop_server.sh restart

### ISSUE THREAD

**Issue by andrewheiss** [getpelican/pelican-plugins#916]
When running `make devserver` on a site that uses `.Rmd` files with figures, the site regenerates because the script notices that figures have been added to or modified in the `figures` directory set in `STATIC_PATHS`, so it re-runs ad infinitum.

For instance, given these settings in `pelicanconf.py`…

    RMD_READER_RENAME_PLOT = 'directory'
    RMD_READER_KNITR_OPTS_CHUNK = {'fig.path': 'figures/'}
    STATIC_PATHS = ['figures']
    PLUGINS = ['rmd_reader']

…and this minimal Rmd saved in `content/pages/testing.Rmd`…

    ---
    title: This is a test
    slug: testing
    date: 2017-07-19
    output: html_document
    ---

    Here's a plot.

    ```{r pressure, echo=FALSE}
    plot(pressure)
    ```

…Pelican generates `content/figures/pages/testing-pressure-1.png`, which is then copied to `output/figures/pages/testing-pressure-1.png` when the site is generated, which is expected.

Everything works great when generating the site once with `pelican content -o output -s pelicanconf.py` or `make html`.

However, when running `make devserver`, which watches for changes in `content/`, the script notices changes to the static `figures` directory and re-generates the site. Over and over and over again:

    -> Modified: [static]figures. re-generating...
    Done: Processed 0 articles, 0 drafts, 6 pages and 0 hidden pages in 1.30 seconds.

    -> Modified: [static]figures. re-generating...
    Done: Processed 0 articles, 0 drafts, 6 pages and 0 hidden pages in 0.64 seconds.

    -> Modified: [static]figures. re-generating...
    Done: Processed 0 articles, 0 drafts, 6 pages and 0 hidden pages in 0.64 seconds.

    -> Modified: [static]figures. re-generating...
    Done: Processed 0 articles, 0 drafts, 6 pages and 0 hidden pages in 0.66 seconds.

Something is changing the `figures` directory, but I can't tell what because everything gets deleted and regenerated so fast and there are no log entries to show what was added or modified. I suspect `rmd_reader` is renaming the generated images based on `RMD_READER_RENAME_PLOT`, which triggers the rebuild.

Is there a way to make `rmd_reader` compatible with `make devserver`?

**Comment by andrewheiss** [comment]
This is likely a @wilsonfreitas question…

**Comment by andrewheiss** [comment]
Ehh, it's likely not because of `RMD_READER_RENAME_PLOT`. The infinite regeneration happens even when `'RMD_READER_RENAME_PLOT': 'disable'`

**Comment by justinmayer** [comment]
@andrewheiss: Were you able to find a solution to your problem here?

**Comment by andrewheiss** [comment]
No, I wasn't. I ended up just using [blogdown](https://bookdown.org/yihui/blogdown/) (R Markdown + Hugo) instead for this project because I couldn't find a way to get Pelican to work.
